src/Silverback.Integration/Messaging/Outbound/Enrichers/BrokerOutboundMessageEnrichersFactory.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Silverback.Messaging.Messages;
+
+namespace Silverback.Messaging.Outbound.Enrichers
+{
+    internal sealed class BrokerOutboundMessageEnrichersFactory : IBrokerOutboundMessageEnrichersFactory
+    {
+        private static readonly NullEnricher NullEnricherInstance = new();
+
+        private readonly IServiceProvider _serviceProvider;
+
+        private readonly ConcurrentDictionary<Type, Type> _enricherTypeCache = new();
+
+        public BrokerOutboundMessageEnrichersFactory(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+        public IEnumerable<IMovePolicyMessageEnricher> GetMovePolicyEnrichers(IEndpoint endpoint)
```

Comment:
Why do you return an `IEnumerable` if you will always only yield a single item?

Reply:
It was because of your previous feedback, but no problem, I changed the implementation.